types/index.d.ts

```diff
@@ -0,0 +1,160 @@
+import type { AedesPacket } from 'aedes-packet';
+import type { IncomingMessage } from 'http';
+import type { Socket } from 'net';
+import type { Duplex, Readable } from 'stream';
+import type { EventEmitter } from 'events';
+
+export type { AedesPacket as Packet } from 'aedes-packet';
+
+export interface AedesPersistenceSubscription {
+  clientId?: string;
+  topic: string;
+  qos: number;
+}
+
+export interface Brokers {
+  [brokerId: string]: {
+    brokerId: string;
+  };
+}
+
+type Connection = Duplex | Socket;
+
+interface Client extends EventEmitter {
```

Comment:
Cannot we use aedes client here instead?

Reply:
In fact that's what i did in my projects, and initially i didn't want to add another dependency in aedes-persistence.
But i'm ok to add it.

Reply:
just add aedes types as dev dep